gping/src/main.rs

```diff
@@ -151,13 +152,18 @@ impl App {
         // Find the Y axis bounds for our chart.
         // This is trickier than the x-axis. We iterate through all our PlotData structs
         // and find the min/max of all the values. Then we add a 10% buffer to them.
-        let iter = self
+        let (min, max) = match self
             .data
             .iter()
             .flat_map(|b| b.data.as_slice())
-            .map(|v| v.1);
-        let min = iter.clone().fold(f64::INFINITY, |a, b| a.min(b));
-        let max = iter.fold(0f64, |a, b| a.max(b));
+            .map(|v| v.1)
+            .minmax()
+        {
+            MinMaxResult::NoElements => unreachable!(),
```

Comment:
There is a slight difference here: in the code you are replacing it returns “INFINITY” and 0 for max and min. Can you return that here?

Reply:
I’m not sure if it ever actually needed, but it’s good to keep compatibility In case it is. If it’s not then let me know, but adding a message to “unreachable!” specifying that the iterator is empty would be handy

Reply:
Sure! It's my mistake, I didn't consider this situation. For consistency, I would keep the return value the same as the old version. I also agree with you, this situation might be considered a bug if it actually reached. This is what pattern match can tell us, and make every case clearer. 

For now, we can keep it as the old style here, because this code would be easy to edit again if there would be some issue in the future.